test/e2e/storage/vsphere/vsphere_utils.go

```diff
@@ -399,6 +411,38 @@ func verifyVSphereVolumesAccessible(c clientset.Interface, pod *v1.Pod, persiste
 	}
 }
 
+// verify volumes are created on one of the specified zones
+func verifyVolumeCreationOnRightZone(persistentvolumes []*v1.PersistentVolume, nodeName string, zones []string) {
+	for _, pv := range persistentvolumes {
+		volumePath := pv.Spec.VsphereVolume.VolumePath
+		// Extract datastoreName from the volume path in the pv spec
+		// For example : "vsanDatastore" is extracted from "[vsanDatastore] 25d8b159-948c-4b73-e499-02001ad1b044/volume.vmdk"
+		datastoreName := volumePath[strings.Index(volumePath, "[")+1 : strings.Index(volumePath, "]")]
```

Comment:
You can use govmomi's `object.DatastorePath.FromString()` to parse VolumePath.

Reply:
Thanks for pointing this out. Making this change.